guides/hack/11-built-in-types/28-shapes.md

```diff
@@ -95,6 +95,19 @@ function takes_server(Server $s) {
 }
 ```
 
+While you can't instantiate these types directly, you can create a my_function_arguments_t by simply defining a shape that has all of the required fields. The typechecker will automatically infer if your shape can be accepted by code() or not.
+
+``` Hack
+$args = shape('name' => 'hello', 'age' => 10);
+$output = takes_server($args); // no error
+
+$args = shape('name' => null, 'age' => 10);
+$output = takes_server($args); // typechecker error: type mismatch
+
+$args = shape('name' => null, 'age' => 10, 'error' => true);
```

Comment:
replace with a string name so there's one error at a time, rather than both the type mismatch and extra field errors